pkg/helpers/klusterletconfig.go

```diff
@@ -0,0 +1,32 @@
+package helpers
+
+import (
+	"fmt"
+
+	listerklusterletconfigv1alpha1 "github.com/stolostron/cluster-lifecycle-api/client/klusterletconfig/listers/klusterletconfig/v1alpha1"
+	klusterletconfighelper "github.com/stolostron/cluster-lifecycle-api/helpers/klusterletconfig"
+	klusterletconfigv1alpha1 "github.com/stolostron/cluster-lifecycle-api/klusterletconfig/v1alpha1"
+	"github.com/stolostron/managedcluster-import-controller/pkg/constants"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+func GetMergedKlusterletConfigWithGlobal(
+	klusterletconfigName string,
+	kcLister listerklusterletconfigv1alpha1.KlusterletConfigLister,
+) (*klusterletconfigv1alpha1.KlusterletConfig, error) {
+	var err error
+	var kc *klusterletconfigv1alpha1.KlusterletConfig
+	if klusterletconfigName != "" {
+		kc, err = kcLister.Get(klusterletconfigName)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return nil, fmt.Errorf("failed to get klusterletconfig %s: %v", klusterletconfigName, err)
+		}
+	}
+
+	globalKlusterletConfig, err := kcLister.Get(constants.GlobalKlusterletConfigName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, fmt.Errorf("failed to get global klusterletconfig: %v", err)
+	}
```

Comment:
If "not found" return kc directly without run merge. (@zhiwei)

Reply:
The `klusterletconfighelper.MergeKlusterletConfigs` implement the process to filterout the nil items, it can save us some code from the client side.

Reply:
I want in the future when we want to implement a real multiple kc merge process, we also don't need to process each kc is nil, just put all kcs as the input and the merge function will handle it.